discussions-api/src/main/java/org/xwiki/contrib/discussions/DiscussionReferencesSerializer.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.discussions;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.contrib.discussions.domain.references.AbstractDiscussionReference;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Perform a serialization of any {@link AbstractDiscussionReference}.
+ *
+ * @version $Id$
+ * @since 2.0
+ */
+@Unstable
+@Role
+public interface DiscussionReferencesSerializer
+{
+    /**
+     * Serialize the given reference to store it as a string.
+     *
+     * @param discussionReference the reference to serialize.
+     * @return a serialization of the reference.
+     */
+    String serialize(AbstractDiscussionReference discussionReference);
```

Comment:
Why not one method per kind of reference? Unless we plan to make it open to others to provide their own reference kinds?

Reply:
So for the serialization it's actually not needed, you can serialize all the `AbstractDiscussionReference` the same way, so I don't see the point of adding complexity to chose the right method for serializing.